### ISSUE TITLE
skopeo don't use user's registries.conf

### ISSUE THREAD

**Issue by lfarkas** [containers/skopeo#800]
even though in my ~/.config/containers/registries.conf i list my own 
```
[registries.insecure] 
registries = ['registry.int.vidux.hu:5000']
```
skopeo still gives this error:
`FATA[0000] Error parsing image name "docker://registry.int.vidux.hu:5000/vidux/live555:latest": error pinging docker registry registry.int.vidux.hu:5000: Get https://registry.int.vidux.hu:5000/v2/: http: server gave HTTP response to HTTPS client`

**Comment by TomSweeneyRedHat** [comment]
@lfarkas At first blosh, it looks as though your registry didn't get setup to handle HTTPS requests, only HTTP.  When you started your registry, did you start it with pointers to a TLS cert and key similar to:

```
docker run -d -p 5000:5000 --name registry -v /root/auth:/root/auth:Z -e "REGISTRY_AUTH=htpasswd" -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" -e REGISTRY_AUTH_HTPASSWD_PATH=/root/auth/htpasswd -e REGISTRY_HTTP_TLS_CERTIFICATE=/root/auth/domain.crt -e REGISTRY_HTTP_TLS_KEY=/root/auth/domain.key registry:2
```

The other thought is did you try running your command but dropping the ':5000' part of the registry?  That should work, but maybe a hiccup there.

**Comment by lfarkas** [comment]
yes, that was my question. there is no https only http in the registry, it's an internal http only registry. that's why we put it into the insecure registry list. and anyway it's working with buildah and podman only skopeo fail with it (since do not read the registries.conf).

**Comment by mtrmac** [comment]
The configuration looks correct; the problem is that  ~/.config/containers/registries.conf is not a standard file location used by default (apparently Podman/Buildah add an explicit override for that, and _only_ in rootless mode).

There is a (hidden?) `skopeo --registries-conf=…` option that should work to point at your configuration file.

**Comment by lfarkas** [comment]
> The configuration looks correct; the problem is that ~/.config/containers/registries.conf is not a standard file location used by default (apparently Podman/Buildah add an explicit override for that, and _only_ in rootless mode).

wasn't that was the main goal of these tools to be able to use in rootless mode?
in this case shouldn't have to support rootless mode and users be able to modify registries?
so IMHO it's a bug.

**Comment by mtrmac** [comment]
Skopeo does not create containers, so it it ~inherently rootless. We’ve, IIRC, just never _had_ the discussion about a per-user registries configuration file as a generic concept, outside of a Podman rootless-specific adjustment.

**Comment by lfarkas** [comment]
but skopeo should have to use/work with rootless local containers so should have to use all of the rootless configuration files in ~/.config/containers/

**Comment by mtrmac** [comment]
The command you are using does not refer to “containers” at all; skopeo does not deal with containers. Images are just blobs of data, they don’t really have a “rootless” concept.

I’m not saying that having a per-user configuration that could be set up per-user without the machine administrator’s involvement is completely unreasonable (and, again, `skopeo --registries-conf` already makes that possible, even if cumbersome); it’s the rootless connection that doesn’t make much sense to me.

**Comment by lfarkas** [comment]
ok than dare i said that skopeo should have to deal with images which stored in a rootless user's home directory. so in this case it's really have to deal the location of the images and also the registry references in rootless mode. and if all other https://github.com/containers/ tools use the ~/.config/containers/ directory for configuration should not skopeo have to do the same?

**Comment by rhatdan** [comment]
I agree skopeo should add support for looking in that location by default.  Would you like to open a PR?

**Comment by mtrmac** [comment]
Not Skopeo; the per-user configuration path, if any, should be defined in c/image (regardless of whether the user is unprivileged, and especially whether user namespaces are involved — which is only the case in Skopeo when the `containers-storage:` transport is used), not inconsistently in every individual caller of that library.

**Comment by lfarkas** [comment]
> Not Skopeo; the per-user configuration path, if any, should be defined in c/image (regardless of whether the user is unprivileged, and especially whether user namespaces are involved — which is only the case in Skopeo when the `containers-storage:` transport is used), not inconsistently in every individual caller of that library.

I don't really understand this statement. do you say that those image which are created/generated/pulled etc by podman in rootless mode can NOT and should NOT have to be examined by skopeo?

**Comment by mtrmac** [comment]
I’m saying that if you want the per-user registries.conf to be used everywhere (which is very reasonable and legitimate), the way to do it is to define that file as the canonical location that is used by the shared code, not to add a manual override to every one of these tools under specific non-default conditions (which, BTW, already differ between Buildah and Podman because their “rootless” detection/definition differs).

**Comment by mtrmac** [comment]
(c/image is https://github.com/containers/image , used by all these tools, which actually interprets the contents of the file.)

**Comment by lfarkas** [comment]
I'm still not understand to whom it concern? it was a simple user's bug report (IMHO the current behaviour is a bug). 
or your comment is to the developers? 
or you suggest me to write a issue to c/image and another issue to skopeo to use that common code? 
or???

**Comment by rhatdan** [comment]
Miloslav is saying we should add code to containers/image to search
$HOME/.config/containers/registries.conf and then skopeo, podman, buildah will get it by default rather then all three of them adding this functionality.

**Comment by rhatdan** [comment]
@QiWang19 PTAL

**Comment by mtrmac** [comment]
`~/.config/containers/registries.conf` (and `.config/containers/registries.conf.d`) is supported after #861 .
